module/library/categories/validator/v1.js

```diff
@@ -6,19 +6,20 @@
  */
 
 module.exports = (req) => {
+	let projectsValidator = {
+		create: function () {
+			req.checkBody('externalId').exists().withMessage('externalId is required')
+			req.checkBody('name').exists().withMessage('name is required')
+		},
+		update: function () {
+			req.checkParams('_id').exists().withMessage('required category id')
+		},
+		projects: function () {
```

Comment:
@Prajwal17Tunerlabs we can make this optional
If external category id is passed look for that specific one else return all category